flink-core/src/main/java/org/apache/flink/util/FlinkUserCodeClassLoader.java

```diff
@@ -67,4 +67,9 @@ protected Class<?> loadClassWithoutExceptionHandling(String name, boolean resolv
             throws ClassNotFoundException {
         return super.loadClass(name, resolve);
     }
+
+    @Override
+    public void addURL(URL url) {
```

Comment:
Looks to me this is not needed.

Reply:
This method is called by SafetyNetWrapperClassLoader, so we need it. but we can change public to protected

Reply:
But it is `protected` by default, so we do not need to override it.

Reply:
Due to `SafetyNetWrapperClassLoader` and `URLClassLoader` are not in the same package, so SafetyNetWrapperClassLoader can not call the `addURL` directly.